### PR TITLE
Drone buffs - Add Acidic Salve to drones

### PR DESCRIPTION
**Pull request by XElectricX**
## About The Pull Request

Drones are nearly useless and are only ever used to wait around until the next evolution. Even as an ancient drone you are not going to be doing much difference. I made 2 changes that will hopefully make the drone a viable caste.

- Doubled the rate of plasma transfer. As a drone, even with 1000 plasma in reserves, you can only transfer 50 at a time. It makes barely a dent in most plasma bars, slower than even resting. 100 at a time is more reasonable for the drone, compared to the hivelord's 200 units and near instant transfer as an example.

- Acidic Salve. A new (not really) ability for the drone. It's a much faster version of the shrike's psychic cure, but with drastically lower healing. The numbers are up for debate, but the stats are: 150 plasma, 40 HP per use (brute and fire), 5 second cooldown. If emitting recovery pheromones, it heals a bonus amount determined by the pheromone strength and max health. According to my testing, with ancient drone's heal pheromones (weak), the bonus ranges from a 0.4% to 4%, depending on the max health. Used a young runner and an ancient crusher to draw this conclusion from.

Ancient Queen, Ancient Crusher, Ancient Runner, Young Runner, Young Crusher, Ancient Spitter, Young Spitter
*Ancient Queen, Ancient Crusher, Ancient Runner, Young Runner, Young Crusher, Ancient Spitter, Young Spitter*
The above image is a small test I made to showcase how the 40 HP heal affects each caste. Unlike the shrike's, it is a set number, not a percentage. It has a more noticeable effect on lower health castes like runners. For a young runner, it takes about 3 uses to almost fill the health bar, which is about 15 seconds of healing if not resting. For anything with more health, it takes much longer. Ideally a drone would serve as a backline healer for xenos out of commission and resting.

## Why It's Good For The Game

Drones are sad and deserve to have some use apart from being a bridge to other castes. I'll gladly take any feedback if this is too powerful for drones, or if it is not enough.

## Changelog
:cl:
add: Drones gain Acidic Salve. Small healing ability with a short cooldown.
tweak: Double the rate of plasma transfer for drones.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm [hunk 1]
```diff
@@ -17,3 +17,47 @@
...
+
+	succeed_activate()
+	add_cooldown()
+
+/mob/living/carbon/xenomorph/proc/salve_healing()	//Slight modification of the heal_wounds proc
+	var/amount = 40	//Smaller than psychic cure, less useful on xenos with large health pools
+	if(recovery_aura)	//Leaving in the recovery aura bonus, not sure if it is too high the way it is
+		amount += recovery_aura * maxHealth * 0.008 // +0.8% max health per recovery level, up to +4%
+	adjustBruteLoss(-amount)
+	adjustFireLoss(-amount)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm]:
```suggestion
	adjustFireLoss(-amount)

```

File: code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm [hunk 2]
```diff
@@ -17,3 +17,47 @@
...
+	playsound(target, "alien_drool", 25)
+
+	new /obj/effect/temp_visual/telekinesis(get_turf(target))
+	var/mob/living/carbon/xenomorph/patient = target
+	patient.salve_healing()
+	patient.updatehealth()
+
+	owner.changeNext_move(CLICK_CD_RANGE)
+
+	log_combat(owner, patient, "salved")	//Not sure how important this log is but leaving it in
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm]:
```suggestion
	log_combat(owner, patient, "acid salved")	//Not sure how important this log is but leaving it in
```

File: code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm [hunk 3]
```diff
@@ -17,3 +17,47 @@
...
+	"<span class='xenowarning'>We cover [target] with our rejuvinating goo!</span>")
+	target.visible_message("<span class='xenowarning'>[target]'s wounds are mended by the acid.</span>", \
+	"<span class='xenowarning'>We feel a sudden soothing chill.</span>")
+
+	playsound(target, "alien_drool", 25)
+
+	new /obj/effect/temp_visual/telekinesis(get_turf(target))
+	var/mob/living/carbon/xenomorph/patient = target
+	patient.salve_healing()
+	patient.updatehealth()
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm]:
```suggestion
```

File: code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm [hunk 4]
```diff
@@ -17,3 +17,47 @@
...
+
+	succeed_activate()
+	add_cooldown()
+
+/mob/living/carbon/xenomorph/proc/salve_healing()	//Slight modification of the heal_wounds proc
+	var/amount = 40	//Smaller than psychic cure, less useful on xenos with large health pools
+	if(recovery_aura)	//Leaving in the recovery aura bonus, not sure if it is too high the way it is
+		amount += recovery_aura * maxHealth * 0.008 // +0.8% max health per recovery level, up to +4%
+	adjustBruteLoss(-amount)
+	adjustFireLoss(-amount)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm]:
```suggestion
	adjustFireLoss(-amount, updating_health = TRUE)
```

**Comment by Rohesie** [comment]
I'm all up for giving support xenos support tools, but not sure about giving a T1 heal.

**Comment by psykzz** [comment]
It might be better to make this a melee range tool?

**Comment by XElectricX** [comment]
@psykzz Someone suggested the same thing, will change it.

**Pull request closed without merging**